src/praisonai-agents/praisonaiagents/plugins/manager.py

```
@@ -437,96 +443,221 @@ def get_single_file_plugin(self, name: str) -> Optional[Dict[str, Any]]:
-                else:
-                    logger.warning(f"Entry point {ep.name} is not callable")
-            except Exception as e:
-                logger.error(f"Failed to load plugin from entry point {ep.name}: {e}")
-                
-        return loaded
+            declared = getattr(plugin.info, "hooks", None) or []
+            declared_values = {h.value if hasattr(h, "value") else h for h in declared}
+            if method_name in declared_values:
+                return True
+        except Exception:
+            pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)